watch/watch.go

```diff
@@ -82,6 +82,10 @@ func (w *watchable) Watch() (interface{}, Watch, error) {
 	}
 
 	c := make(chan struct{}, 1)
+	if w.value != nil {
```

Comment:
nit: can we do this outside of the write lock?  

````golang
notify := w.value != nil
w.active = append(w.active, c)
closeFn := w.closeFunc(c)
w.Unlock()

if notify {
   select {
      case c <- struct{}{}:
      default:
   }
}
````

That way we minimize the amount of done with the lock held and reduce contention.